core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -491,21 +489,49 @@ export class OptimizedSsrEngine {
    * the "W3C TraceContext" document. See https://www.w3.org/TR/trace-context/#traceparent-header
    * for more details.
    * @param request - the request object
-   * @returns the context of the request
+   * @returns the context of the request and error if occurred during parsing traceparent header
    * @private
    */
-  private getRequestContext(request: Request): ExpressServerLoggerContext {
+  private getRequestContext(
+    request: Request
+  ): [ExpressServerLoggerContext, Error | null] {
+    let error: Error | null = null;
+    // request.headers['traceparent'] =
+    //   '00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-0';
     const requestContext: ExpressServerLoggerContext = {
       uuid: randomUUID(),
       timeReceived: new Date().toISOString(),
     };
 
-    const traceContext = parseTraceparent(request.get('traceparent'));
-    if (traceContext) {
-      requestContext.traceContext = traceContext;
+    try {
+      const traceContext = parseTraceparent(request.get('traceparent'));
+      if (traceContext) {
+        requestContext.traceContext = traceContext;
+      }
+    } catch (e) {
+      error =
+        e instanceof Error
+          ? e
+          : new Error('Unexpected error during parsing traceparent header');
     }
 
-    return requestContext;
+    return [requestContext, error];
```

Comment:
Did you consider rethrowing the error from this function instead of returning a touple type?
IMHO having custom a touple type here looks quite strange, while try/catch would suffice.
What do you think?